code/modules/mob/living/carbon/alien/humanoid/humanoid.dm

```diff
@@ -65,10 +64,10 @@ GLOBAL_LIST_INIT(strippable_alien_humanoid_items, create_strippable_list(list(
 
 /mob/living/carbon/alien/humanoid/alien_evolve(mob/living/carbon/alien/humanoid/new_xeno)
 	drop_all_held_items()
-	..()
+	return ..()
 
 //For alien evolution/promotion/queen finder procs. Checks for an active alien of that type
-/proc/get_alien_type(var/alienpath)
+/proc/get_alien_type_in_hive(datum/alienpath)
```

Comment:
Should this ensure that the alien belongs to the same hive? For example its not just some random xenobio spawn

Reply:
I don't think we support multiple hives unfortunately